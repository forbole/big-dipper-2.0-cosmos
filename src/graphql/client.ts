import WebSocket from 'isomorphic-ws';
import {
  ApolloClient,
  InMemoryCache,
  split,
  HttpLink,
  ApolloLink,
  concat,
} from '@apollo/client';
import {
  getMainDefinition,
} from '@apollo/client/utilities';
import { WebSocketLink } from '@apollo/client/link/ws';

import { useMemo } from 'react';

const defaultOptions:any = {
  watchQuery: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
  query: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
};

let apolloClient;

const httpLink = new HttpLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_URL,
});

const wsLink = new WebSocketLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_WS ?? 'wss://localhost:3000',
  options: {
    reconnect: true,
  },
  webSocketImpl: WebSocket,
});

const link = typeof window !== 'undefined' ? split(
  ({ query }) => {
    const {
      kind, operation,
    }:any = getMainDefinition(query);
    return kind === 'OperationDefinition' && operation === 'subscription';
  },
  wsLink,
  httpLink,
) : httpLink;

const authMiddleware = new ApolloLink((operation, forward) => {
  operation.setContext({
    headers: {
    },
  });

  return forward(operation);
});

function createApolloClient() {
  const client = new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: concat(authMiddleware, link),
    cache: new InMemoryCache({
    }),
  });

  client.defaultOptions = defaultOptions;

  return client;
}

export function initializeApollo(initialState = null) {
  // eslint-disable-next-line
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();
    // Restore the cache using the data passed from getStaticProps/getServerSideProps
    // combined with the existing cached data
    _apolloClient.cache.restore({
      ...existingCache, ...initialState,
    });
  }
  // For SSG and SSR always create a new Apollo Client
  if (typeof window === 'undefined') return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState) {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
