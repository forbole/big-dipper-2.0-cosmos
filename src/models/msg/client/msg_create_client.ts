import { Categories } from '../types';

class MsgCreateClient {
    public category: Categories;
    public type: string;
    public signer: string;
    public chainId: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.chainId = payload.chainId;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgCreateClient({
        json,
        type: json['@type'],
        signer: json.signer,
        chainId: json.client_state?.chain_id,
      });
    }
}

export default MsgCreateClient;
