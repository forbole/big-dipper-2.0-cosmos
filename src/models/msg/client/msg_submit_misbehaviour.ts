import { Categories } from '../types';

class MsgSubmitMisbehaviour {
    public category: Categories;
    public type: string;
    public signer: string;
    public clientId: string;
    public misbehaviour: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.clientId = payload.clientId;
      this.misbehaviour = payload.misbehaviour;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgSubmitMisbehaviour({
        json,
        type: json['@type'],
        signer: json.signer,
        clientId: json.client_id,
        misbehaviour: json.misbehaviour,
      });
    }
}

export default MsgSubmitMisbehaviour;
