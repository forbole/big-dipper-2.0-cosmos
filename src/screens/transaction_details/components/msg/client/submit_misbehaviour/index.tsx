import React from 'react';
import Trans from 'next-translate/Trans';
import { Typography } from '@material-ui/core';
import { Name } from '@components';
import { MsgSubmitMisbehaviour } from '@models';
import { useChainContext } from '@contexts';

const SubmitMisbehaviour = (props: {
    message: MsgSubmitMisbehaviour;
}) => {
  const { findAddress } = useChainContext();
  const { message } = props;

  const signer = findAddress(message.signer);
  const signerMoniker = signer ? signer?.moniker : message.signer;

  return (
    <Typography>
      <Trans
        i18nKey="message_contents:txSubmitMisbehaviourContent"
        components={[
          (
            <Name
              address={message.signer}
              name={signerMoniker}
            />
          ),
          <b />,
        ]}
        values={{
          clientId: message.clientId,
        }}
      />
    </Typography>
  );
};

export default SubmitMisbehaviour;
