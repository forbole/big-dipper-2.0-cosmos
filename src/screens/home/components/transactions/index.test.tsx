import React from 'react';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import {
  createMockClient, createMockSubscription,
} from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import {
  TransactionsListenerDocument,
} from '@graphql/types/general_types';
import Transactions from '.';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@components', () => ({
  Box: (props) => <div id="Box" {...props} />,
  NoData: (props) => <div id="NoData" {...props} />,
}));

jest.mock('./components', () => ({
  Mobile: (props) => <div id="Mobile" {...props} />,
  Desktop: (props) => <div id="Desktop" {...props} />,
  Box: (props) => <div id="Box" {...props} />,
}));

const mockTxsListenerDocument = {
  data: {
    transactions: [
      {
        height: 2000,
        timestamp: '2021-02-18T09:02:28.668623',
        hash: '76nwV8zz8tLz97SBRXH6uwHvgHXtqJDLQfF66jZhQ857',
        messages: 12,
        success: true,
      },
    ],
  },
};

// ==================================
// unit tests
// ==================================
describe('screen: Home/Blocks/Mobile', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    const mockSubscription = createMockSubscription();

    mockClient.setRequestHandler(
      TransactionsListenerDocument,
      () => mockSubscription,
    );

    let component;

    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <Transactions />
          </MockTheme>
        </ApolloProvider>,
      );
    });

    await wait();

    renderer.act(() => {
      mockSubscription.next(mockTxsListenerDocument);
    });

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
