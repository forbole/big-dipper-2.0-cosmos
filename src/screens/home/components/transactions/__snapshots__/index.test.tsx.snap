// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/Blocks/Mobile matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Box"
>
  <div
    className="makeStyles-label"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2"
    >
      latestTransactions
    </h2>
    <a
      className="MuiTypography-root button MuiTypography-h4"
      href="/transactions"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      seeMore
    </a>
  </div>
  <div
    id="NoData"
  />
</div>
`;
